homeassistant/components/sensor/netatmo.py

```diff
@@ -313,3 +321,131 @@ def update(self):
                 station=self.station, exclude=3600)
         else:
             self.data = self.station_data.lastData(exclude=3600)
+
+        newinterval = 0
+        for module in self.data:
+            if 'When' in self.data[module]:
+                newinterval = self.data[module]['When']
+                break
+        if newinterval:
+            # Try and estimate when fresh data will be available
+            newinterval += NETATMO_UPDATE_INTERVAL - time()
+            if newinterval >= NETATMO_UPDATE_INTERVAL - 30:
+                newinterval = NETATMO_UPDATE_INTERVAL
+            elif newinterval < NETATMO_UPDATE_INTERVAL / 2:
+                # Never hammer the NetAtmo API more than
+                # twice per update interval
+                newinterval = NETATMO_UPDATE_INTERVAL / 2
+            _LOGGER.warning("NetAtmo refresh interval reset to %d seconds",
+                            newinterval)
+            self.update = Throttle(timedelta(seconds=newinterval),
```

Comment:
I think that it is getting overly complicated with the throttle.

Why not just keep track of a timestamp and that's it?

```python
if self._call_in_progress or dt_util.utcnow() < this._next_call:
    return

self._call_in_progress = True
try:
    self.next_call = newinterval
    # All the other fechting stuff
finally:
    self._call_in_progress = False
```

Reply:
Surely the throttle has its intrinsic elegance ;-) but yes, this is a good solution as well, and it avoids the code duplication. I'll look into that.